Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java

```diff
@@ -62,7 +56,27 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
     @Override
     public void onResume() {
         super.onResume();
-        payPalLauncher.handleReturnToAppFromBrowser(requireContext(), requireActivity().getIntent());
+        PayPalPendingRequest.Started pendingRequest = getPendingRequest();
```

Comment:
```suggestion
        PayPalPendingRequest pendingRequest = getPendingRequest();
```
nit: Would it be the base class here since it could technically be `.Started` or `.Failure`?

Reply:
Our docs tell merchants to only store a `PayPalRequest.Started`, so that's what the persistent store returns here. If the pending request was `PayPalPendingRequest.Failure` there is no need to store it, because the user can't complete that flow anyways, so they should handle an error immediately after calling `start`. Does that seem unclear for merchants?

Reply:
Oh got it yeah that makes sense. I'm leaning toward a `PayPalLaunchResult.Success` type with a `PayPalPendingRequest` associated value but re-using the `PayPalRequest.Started` type also makes sense.

I'm also thinking from a PPCP perspective, the requirements are bit different on our side.

Reply:
Since they can't pass it back into the `parseResult` method (which only accepts a `PayPalPendingRequest.Started`) I think while setting up the code it would become clear not to store a `Failure`, but the benefit of returning a `Failure` rather than `null` is that they can get the Exception `cause` from the `Failure` for more info, where they can't with a null response.